File: problems/binary_search/PR_43238_입국심사.py
def solution(n, times):
    # mid : 총 걸린 시간.
    start = min(times)
    end = n * max(times)
    answer = 0
    
    while start <= end:
        count = 0
        mid = (start + end) // 2
        for time in times:
            count += mid // time

        if count < n:
            start = mid + 1
        elif count >= n:
            answer = mid
            end = mid - 1

    return answer

File: problems/binary_search/PR_43238_입국심사.js
function solution(n, times) {  
  let start = 0
  let end = times[times.length - 1] * n
  let mid = 0
  let answer = end
  while(start <= end) {
      mid = Math.floor((start + end) / 2)
      let available = times.reduce((acc, cur) => acc + Math.floor(mid / cur), 0)
      
      if (available >= n) {
          answer = Math.min(answer, mid)
          end = mid - 1
      } else {
          start = mid + 1
      }
      
  }
  
  return answer;
}